exec/cnex/lib/os_posix.c

```diff
@@ -0,0 +1,192 @@
+// The following is needed for kill() on Posix platforms.
+#define _POSIX_C_SOURCE 200809L
+
+#include <assert.h>
+#include <errno.h>
+#include <signal.h>
+#include <stdlib.h>
+#include <sys/param.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+
+#include "cell.h"
+#include "enums.h"
+#include "exec.h"
+#include "number.h"
+#include "object.h"
+#include "stack.h"
+
+
+
+static Array *g_children;
+
+typedef struct tagTProcessHandle {
+    pid_t proc;
+} TProcessHandle;
+
+
+
+static TProcessHandle *check_process(TExecutor *exec, Object *pp)
+{
+    if (pp == NULL || pp->ptr == NULL || ((TProcessHandle*)pp->ptr)->proc == 0) {
+        exec->rtl_raise(exec, "OsException.InvalidProcess", "", BID_ZERO);
+        return NULL;
+    }
+    return pp->ptr;
+}
+
+void object_releaseProcessObject(Object *o)
+{
+    if (o != NULL) {
+        assert(o->refcount > 0);
+        o->refcount--;
+
+        if (o->refcount <= 0) {
+            if (o->ptr != NULL) {
+                free(o->ptr);
+            }
+            free(o);
+        }
+    }
+}
+
+Cell *object_processObjectToString(Object *self)
+{
+    Cell *r = cell_fromCString("<PROCESS ");
+    char s[16];
+    snprintf(s, 16, "%d>", ((TProcessHandle*)self->ptr)->proc);
```

Comment:
Rather than doing this piecemeal, you could just snprintf the whole thing (`"<PROCESS %d>"`) into a buffer and then return a cell from that without having to append anything.

Reply:
Done.  And both of them still fit into a 32 byte buffer too.